packages/demo/src/components/examples/DatePickerExamplesCommon.ts

```diff
@@ -154,6 +154,21 @@ export const CALENDAR_SELECTION_RULES: ICalendarSelectionRule[] = [
     },
 ];
 
+export const CALENDAR_ADVANCED_SELECTION_RULES: ICalendarSelectionRule[] = [
+    {
+        test: (date: Date) => date <= new Date() && date >= moment().subtract(2, 'weeks').toDate(), // is the date within the last two weeks
```

Comment:
should those rules have the range allowed as parameter?
I feel like we would need to add more rules for use case where the range is 3 days, 1 day, etc :thinking:

Reply:
Also, I think we should have a description popping at the same time that the red border if we are going down that route :thinking: 
Right now, it turns red but we don't know why (well we do but our users probably won't :see_no_evil: )

Reply:
Maybe mimic this pattern instead? :eyes: 
![image](https://user-images.githubusercontent.com/63734941/140079023-cf515f8b-a450-40b6-b52f-8a6412899457.png)

Reply:
> should those rules have the range allowed as parameter? I feel like we would need to add more rules for use case where the range is 3 days, 1 day, etc 🤔

That was my first attempt but using range triggers the infinite loop

> Also, I think we should have a description popping at the same time that the red border if we are going down that route 🤔
Right now, it turns red but we don't know why (well we do but our users probably won't 🙈 )

I thought that too; I didn't attempt it yet as I wasn't sure what I have so far was the right way to go, and adding a message involves a fair bit more complexity